### ISSUE TITLE
Nested root o2m modifiers (some / every)

### ISSUE THREAD

**Issue by rijkvanzanten** [directus/directus#19932]

**Comment by Brian-Hofmann** [comment]
I think this may be what I am looking for. I am trying to make a filter that checks if every value I pass in an array ["value1", "value2"] is in the many to many relationship of my collection. So for example, if I have a book with many authors (authors and books having a many to many relationship), I would want to pass in a filter value which is an array of author names and only get books which contain **_every_** author instead of just one. 

Is that not currently supported and would be once this issue is resolved?

**Comment by jaads** [comment]
Exactly! For this use case you would need the `every` operator we want to implement here. But we will add this in the new data abstraction (called Durus) instead of in the current Knex based implementation. So you will only be able to use it as soon as this project is done and integrated in Directus. Since this is a biig project, it will take some time.

**Comment by nickrum** [comment]
@Brian-Hofmann If I understand it correctly, what you are after is "give me the books where _some_ author's first name is John and _some_ author's first name is Steve". This can already be done in Directus like this:

```json
"_and": {
  "books": {
    "authors": {
      "author": {
        "first_name": {
          "_eq": "John"
        }
      }
    }
  },
  "books": {
    "authors": {
      "author": {
        "first_name": {
          "_eq": "Steve"
        }
      }
    }
  }
}
```

There is no array syntax for this, though. The reason being that Directus implicitly quantifies the part of the filter that is within a o2m using _some_. More on this can be found [here](https://docs.directus.io/reference/filter-rules.html#some-vs-none-in-one-to-many). The more explicit version of your filter would look like this:

```json
"books": {
  "_some": {
    "authors": {
      "author": {
        "first_name": {
          "_every": ["John", "Steve"]
        }
      }
    }
  }
}
```

This in turn means that the part of the filter within the o2m can only filter based on a single item in authors. Think of it as a loop that goes through each author individually and thus you have to use an outside _and_.

As already mentioned by @jaads, this issue is about implementing this _some_ quantifier, that is already part of Directus (and a new one that is called _every_ which replaces the current _none_ quantifier), in a future project.

**Comment by Brian-Hofmann** [comment]
Perfect. I'll track that open issue then. Thanks!
